d3-dag.d.ts

```diff
@@ -0,0 +1,781 @@
+             * simply reverse the DAG post creation. */
+            reverse(): this
+            /** Set the value of each node to be the number
+             * of descendants including itself */
+            count(count: number): this
+            /** Returns true if the dag is connected. */
+            connected(): boolean
+            /** Set the value of each node to be zero if its a root node,
+             * or the greatest distance to any root node for other nodes. */
+            depth(): this
+            /** Set the value of each node to be zero if its a leaf node,
+             * or the greatest distance to any leaf node for other nodes. */
+            height(): this
+            /** Invoke the specified function on each node in an arbitrary order. */
+            each(callback: (node: this) => void): this
+            /** Invoke the specified function on each node such a node is called before any of its parents. */
+            eachAfter(callback: (node: this) => void): this
+            /** Invoke the specified function on each node such a node is called before any of its children. */
+            eachBefore(callback: (node: this) => void): this
+            /** Invoke the specified function on each node in breadth first order. */
+            eachBreadth(callback: (node: this) => void): this
+            /** Return true if this dag is equal to that dag.
+             *  For two dags to be equal the data must be strictly (===) equal. */
+            equals(that: this): boolean
+            /** Return true if function returns true for every node in the DAG. */
+            every(callback: (node: this) => boolean): boolean
+            /** Return true if function returns true for at least one node in the DAG. */
+            some(callback : (node: this) => boolean): boolean
+            /** Set the value of every node to be the sum of this functions return value
+             * on the current node's data and the value of every descendant's return value. */
+            sum(callback : (node: this) => number): this
+            /** return number of nodes */
+            size(): number
+            /** Calls the specified callback function for all nodes. The return value of the callback function is the accumulated result, and is provided as an argument in the next call to the callback function.
+             * @param callbackfn A function that accepts up to four arguments. The reduce method calls the callbackfn function one time for each element in the array.
+             * @param initialValue If initialValue is specified, it is used as the initial value to start the accumulation. The first call to the callbackfn function provides this value as an argument instead of an array value.
+            */
+            reduce<U>(callbackfn: (previousValue: U, currentValue: this, currentIndex: number) => U, initialValue: U): U;
+        }
+
+        /** Link between 2 Nodes */
+        export interface Link<DAGNode, LinkDatum> {

```

Comment:
nit: I find most style guides recommend acronyms not get all capitalized. I think it's mostly for appearance, but I would also probably argue `DagNode` conveys what it is better.